### PR TITLE
Add iteration for Cholesky and CholeskyPivoted

### LINKED ISSUE #739: LinearAlgebra - Cholesky decomposition iteration
Iteration is not implemented for Cholesky decomposition.

Implementing iteration will make it possible to iterate over the Cholesky factors of a matrix:

`L, U = cholesky(A)`

I'd like to submit a pull request

### LINKED ISSUE #36520: LinearAlgebra - Cholesky decomposition iteration
Iteration is not implemented for Cholesky decomposition.

Implementing iteration will make it possible to iterate over the Cholesky factors of a matrix:

`L, U = cholesky(A)`

I'd like to submit a pull request

### PR DESCRIPTION
**Pull request by axmat**
Fixed JuliaLang/LinearAlgebra.jl#739

### CODE REVIEW

File: stdlib/LinearAlgebra/src/cholesky.jl [hunk 1]
```diff
@@ -67,6 +67,11 @@ julia> C.L
 
 julia> C.L * C.U == A
 true
+
+julia> l, u = C; # destructuring via iteration
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/cholesky.jl:71]: You should also add a comment to the above docstring à la
```
Iterating the decomposition produces the components `L` and `U`.
```
Similarly below.

**Comment by dkarrasch** [comment]
@axmat Sorry for the delay. This looks good. Could you also please add some tests for it, basically exactly what you have in the `jldoctest`, but this time in the `tests`. After that, this is ready to go.

**Comment by axmat** [comment]
@dkarrasch have a look. I think it's ready to be merged.

File: stdlib/LinearAlgebra/test/cholesky.jl [hunk 2]
```diff
@@ -429,4 +429,22 @@ end
     @test cholstring == "$(summary(C))\n$rankstring\n$factorstring\npermutation:\n$permstring"
 end
 
+@testset "Issue #36520 iteration for Cholesky" begin
+    A = rand(8,8)
+    B = A'A
+    C = cholesky(B, Val(false), check=false)
+    l, u = C
+    @test l == C.L 
+    @test u == C.U
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/cholesky.jl:438]:
```suggestion
    @test l == C.L
    @test u == C.U
```

File: stdlib/LinearAlgebra/test/cholesky.jl [hunk 3]
```diff
@@ -429,4 +429,22 @@ end
...
+    @test u == C.U 
+end
+
+@testset "Issue #36520 iteration for CholeskyPivoted" begin
+    A = rand(8,8)
+    B = A'A
+    C = cholesky(B, Val(true), check=false)
+    l, u = C
+    @test l == C.L
+    @test u = C.U
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/cholesky.jl:447]:
```suggestion
    @test u == C.U
```

**Comment by dkarrasch** [comment]
Thanks! Actually, I'd suggest to merge the two tests. Like
```julia
@testset "destructuring of Cholesky*" begin
    for val in (true, false)
        A = rand(8, 8)
        B = A'A
        C = cholesky(B, Val(val), check=false)
        l, u = C
        @test l == C.L
        @test u == C.U
    end
end
```

File: stdlib/LinearAlgebra/test/cholesky.jl (hunk 2 above)
Comment (axmat) [reply stdlib/LinearAlgebra/test/cholesky.jl:438]: Thanks

File: stdlib/LinearAlgebra/test/cholesky.jl (hunk 3 above)
Comment (axmat) [reply stdlib/LinearAlgebra/test/cholesky.jl:447]: Thanks

**Review by dkarrasch — APPROVED** [review]

**Comment by dkarrasch** [comment]
Thanks, @axmat, and welcome to the Julia contributors. 😉

**Pull request merged**